InvalidOperationException: Sequence contains no matching element in CollectDeclaredReferencesTask

Saw this when opening Gardener.sln without having it restored

```
C:\Users\kirillo\.nuget\packages\referencetrimmer\3.3.6\build\ReferenceTrimmer.targets (31,5): error MSB4018: The "CollectDeclaredReferencesTask" task failed unexpectedly.
System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at ReferenceTrimmer.Tasks.CollectDeclaredReferencesTask.GetPackageInfos() in D:\a\ReferenceTrimmer\ReferenceTrimmer\src\Tasks\CollectDeclaredReferencesTask.cs:line 242
   at ReferenceTrimmer.Tasks.CollectDeclaredReferencesTask.Execute() in D:\a\ReferenceTrimmer\ReferenceTrimmer\src\Tasks\CollectDeclaredReferencesTask.cs:line 157
   at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

--- a/src/Tasks/CollectDeclaredReferencesTask.cs
+++ b/src/Tasks/CollectDeclaredReferencesTask.cs
@@ -254,9 +254,15 @@ public sealed class CollectDeclaredReferencesTask : MSBuildTask
             }
 
             string nugetLibraryRelativePath = lockFile.GetLibrary(nugetLibrary.Name, nugetLibrary.Version).Path;
-            string nugetLibraryAbsolutePath = packageFolders
+            string? nugetLibraryAbsolutePath = packageFolders
                 .Select(packageFolder => Path.Combine(packageFolder, nugetLibraryRelativePath))
-                .First(Directory.Exists);
+                .FirstOrDefault(Directory.Exists);
+            if (nugetLibraryAbsolutePath is null)
+            {
+                // This can happen if the project has a stale lock file.
+                // Just ignore it as NuGet itself will likely error.
+                continue;
+            }
 
             List<string> nugetLibraryAssemblies = nugetLibrary.CompileTimeAssemblies
                 .Select(item => item.Path)

